packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js

```diff
@@ -182,4 +182,44 @@ describe('BlocksEditor toolbar', () => {
       },
     ]);
   });
+
+  it('transforms the selection to a quote when selected and trasforms it back to text when selected again', async () => {
+    setup();
+
+    const headingsDropdown = screen.getByRole('combobox', { name: /Select a block/i });
+
+    Transforms.setSelection(baseEditor, {
+      anchor: { path: [0, 0], offset: 0 },
+    });
+
+    await user.click(headingsDropdown);
+
+    await user.click(screen.getByRole('option', { name: 'Quote' }));
+
+    expect(baseEditor.children).toEqual([
+      {
+        type: 'quote',
+        children: [
+          {
+            text: 'A line of text in a paragraph.',
```

Comment:
Could you please explain necessity of writing type:'text' for the text nodes.
I thought only text is enough to determine leaf nodes, [Slate](https://docs.slatejs.org/api/nodes/text)

Reply:
Yes that is true Slate only requires the `text`. In the [RFC](https://www.notion.so/strapi/New-rich-text-editor-RFC-9c85b3fb1a2f4c6b844a01daea4ffd98?pvs=4#114bdd64fb26490eb33b10a97a3d67e1) we defined a Strapi-specific schema for the nodes in the rich text tree. All nodes have a type property. Over [on my PR](https://github.com/strapi/strapi/pull/18079/files#diff-ca00dda7b24eb88429e1116532b8034076e52cfcfd79da58f8fdcbf9669566ef) I setup the validation based on the schema we defined in the RFC so if you try to save a node that doesn't respect the schema it will throw an error.